aten/src/ATen/core/LazyTensor.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <c10/core/TensorImpl.h>
+#include <ATen/core/ivalue.h>
+
+namespace at {
+
+struct CAFFE2_API LazyTensorImpl : public TensorImpl {
+  explicit LazyTensorImpl(
+      const caffe2::TypeMeta& data_type,
+      c10::Device device_opt)
+      : TensorImpl(TensorTypeId::LazyTensorId, data_type, device_opt),
```

Comment:
I've been thinking about whether, in the "type set" world, lazy tensors should also have dispatch keys for, e.g., variable and cpu/cuda tensor.

On the side of yes: tensor type sets are currently used for instanceof tests, and in transparent lazy usage, you want lazy cpu tensors to report as cpu tensors

On the side of no: tensor type sets are used to make claims about the *structure* of the TensorImpl; but a lazy cpu tensor "is not a" cpu tensor.

@zdevito Maybe this means we really should organize one of these notions some other way on Tensor.